src/components/DiffView/index.tsx

```diff
@@ -6,28 +6,48 @@ import {
   DiffProps,
   Hunk,
   HunkInfo,
+  getChangeKey,
   parseDiff,
   tokenize,
 } from 'react-diff-view';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
 
 import refractor from '../../refractor';
 import { getLanguageFromMimeType } from '../../utils';
 import styles from './styles.module.scss';
 import 'react-diff-view/style/index.css';
 
-type PublicProps = {
+export type PublicProps = {
+  _document: typeof document;
   _tokenize: typeof tokenize;
   diff: string;
   mimeType: string;
   viewType: DiffProps['viewType'];
 };
 
-class DiffView extends React.Component<PublicProps> {
+type Props = PublicProps & RouteComponentProps;
+
+export class DiffViewBase extends React.Component<Props> {
   static defaultProps = {
+    _document: document,
     _tokenize: tokenize,
-    viewType: 'unified',
+    viewType: 'unified' as PublicProps['viewType'],
   };
 
+  componentDidMount() {
+    const { _document, location } = this.props;
+
+    if (!location.hash.length) {
+      return;
+    }
+
+    const element = _document.querySelector(location.hash);
+
+    if (element) {
+      element.scrollIntoView();
```

Comment:
This seems fine for the first mount but what about scrolling when you click on a line? Could you file a follow-up issue to address that? However, I'm not sure why `react-diff-view` doesn't already scroll the page whenever you click on a line. Maybe its anchor tag click handler is returning false.

Reply:
o_O Do you mean scrolling does not work for you? I believed it scrolled unless  the bottom of the file reaches the bottom of the page.

Reply:
It works for me, but because the diff is small, only the first lines scroll.

Reply:
> because the diff is small, only the first lines scroll

Ah, yes, this is what was leading me to believe it was not working. It is working 🎉